railties/test/application/loading_test.rb

```diff
@@ -139,9 +139,11 @@ class Post < ApplicationRecord
 
     assert_equal initial, ActiveRecord::Base.descendants.collect(&:to_s).sort.uniq
     get "/load"
-    assert_equal [Post].collect(&:to_s).sort, ActiveRecord::Base.descendants.collect(&:to_s).sort - initial
+    assert_equal 200, last_response.status
```

Comment:
In a very interesting cascade of events, turns out the failing test was actually because this was returning a 500 due to a `table 'posts' not found`!! The table was not being found because the new connection switching code in the `check_pending_migrations` call re-establishes the `Base.connection`, even if there is only a single database configured, which is usually fine, except the `setup_ar!` method of this test file was for some reason overriding the SQLite database name to `:memory:`, which lo and behold, means getting a fresh/empty database on every reconnect (essentially dropping the table by the time we actually run the request).

> Instead, a new database is created purely in memory. The database ceases to exist as soon as the database connection is closed. Every :memory: database is distinct from every other. So, opening two database connections each with the filename ":memory:" will create two independent in-memory databases.

(see https://www.sqlite.org/inmemorydb.html)

🤯

Reply:
Anyway, I have improved a few issues I saw with the assertions in this test as well. Namely, 

1. asserting that the last request actually succeeded 
2. not referencing `Post` from outside the app, because otherwise our assertion would be invalid

Reply:
Just so we can be sure this is unrelated to your changes, do you mind sending these changes as a separate PR that I'll merge first?

Reply:
https://github.com/rails/rails/pull/46237